src/shogun/base/SGObject.cpp

```diff
@@ -254,6 +254,7 @@ void CSGObject::update_parameter_hash()
 	uint32_t carry=0;
 	uint32_t length=0;
 
+	m_hash=0;
```

Comment:
@karlnapf 
the bug is mentioned at
https://github.com/shogun-toolbox/shogun/issues/2838